File: Programmers/level1/level1_폰켓몬.py
def solution(nums):
    N = len(nums) // 2
    items = dict()
    
    for n in nums:
        if n not in items:
            items[n] = 0
        items[n] += 1
    
    answer = N if N < len(items) else len(items)
    return answer

File: Programmers/level1/level1_폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int N = nums.length / 2;
        Map<Integer, Integer> map = new HashMap<>();
        for (int num : nums) {
            if (!map.containsKey(num)) {
                map.put(num, 0);
            }
            map.put(num, map.get(num) + 1);
        }
        return N <= map.size() ? N : map.size();
    }
}

